### PR TITLE
Add csurf

### LINKED ISSUE #15: Fix code scanning alert - Missing CSRF middleware
Tracking issue for:
- [ ] https://github.com/BranAndSceolan/SSE-Notes/security/code-scanning/1

### PR DESCRIPTION
**Pull request by BranAndSceolan**
Add csurf to protect against csrf.
Also try to use csurf in tests to make finding mistakes more likely.

### CODE REVIEW

File: backend/index.ts [hunk 1]
```diff
@@ -24,23 +27,37 @@
...
+} else {
+    app.use(session({
+        resave: true, // save session even if not modified
+        saveUninitialized: true, // save session even if not used
+        rolling: true, // forces cookie set on every response needed to set expiration
+        secret: crypto.randomInt(0, 1000000).toString(), // encrypt session-id in cookie using "secret" as modifier
+        name: "myawesomecookie", // name of the cookie set is set by the server
+        cookie: {maxAge: 15*60*1000}, //enable this as soon as https-certificates are included, and we use https for our messages
+        // only then will this application be secure!
+    }));
```
[the remark this answers is not in the document]
Comment (BranAndSceolan) [reply backend/index.ts]: cookies will get added protection when we add https

File: backend/index.ts [hunk 2]
```diff
@@ -25,30 +38,40 @@
...
+    app.use(session({
+        resave: true, // save session even if not modified
+        saveUninitialized: true, // save session even if not used
+        rolling: true, // forces cookie set on every response needed to set expiration
+        secret: crypto.randomInt(0, 1000000).toString(), // encrypt session-id in cookie using "secret" as modifier
+        name: "myawesomecookie", // name of the cookie set is set by the server
+        //TODO: cookie: {secure: true} //enable this as soon as https-certificates are included and we use https for our messages
+        // only then will this application be secure!
+        cookie: {maxAge: 15 * 60 * 1000}
+}))
```
[the remark this answers is not in the document]
Comment (BranAndSceolan) [reply backend/index.ts]: Will address this in another issue

**Comment by BranAndSceolan** [comment]
This request can only be merged into backend when frontend can deal with csrf-tokens.
Otherwise communication between frontend and backend will fail.

**Review by danielp0505 — APPROVED** [review]

**Comment by BranAndSceolan** [comment]
will be included in merge of #90

**Pull request closed without merging**